### ISSUE TITLE
Does singleton attribute work on normal methods?

### ISSUE THREAD

**Issue by mkliu** [Azure/azure-webjobs-sdk#1782]
So I want the entire exe to be singleton. Can I apply the singleton attribute to main function?

**Comment by brettsam** [comment]
No, it can only be applied to Functions and Triggers. What is the scenario that you're trying to accomplish?

**Comment by mkliu** [comment]
Basically a master slave model. At one time, I only want one work to be active. The old singleton model I believe supports that?

When you say "Functions", what's the exact definition? Isn't main function a function?

**Comment by brettsam** [comment]
What do you mean by the "old" singleton model?

Main is a .NET method. By "Functions", I mean WebJobs functions -- functions that are invoked via WebJob Triggers.

**Comment by mkliu** [comment]
@brettsam i mean like {"is_singleton":true} in settings.job file. Is that continue to be supported?
